realm/src/main/java/io/realm/RealmResults.java

```diff
@@ -31,17 +31,19 @@
 import io.realm.internal.TableView;
 
 /**
- * A RealmResults list contains a list of objects of a given type that matches the query.
- * The objects are not copied from the Realm to the RealmResults list, but just references the original objects.
- * This preserves memory and increase speed.
- * It also implies that any modification to any object in a RealmResults is reflected in the objects in the 
- * Realm that was queried.
- * Updates to objects must be done within a transaction and the modified object is persisted to the backing
- * Realm file during the commit of the transaction.
+ * This class holds all the matches of a {@link io.realm.RealmQuery} for a given Realm. The objects
+ * are not copied from the Realm to the RealmResults list, but are just referenced from the
+ * RealmResult instead. This saves memory and increases speed.
```

Comment:
We should mention that they are live views and that they will update automatically if in a Looper thread